testing/e2e/README.md

```diff
@@ -9,3 +11,46 @@ cargo test -p ipc_e2e --test subnet_lifecycle
 ```
 
 > Note: This is a basic skeleton to showcase how we can run automated end-to-end tests over IPC. In the future, the goal is to automate the deployment of the IPC agent and the infrastructure so all tests can be run automatically.
+
+
+## Test environment
+
+The `template` directory contains `docker-compose` files for creating a test environment with varying number of agents and nodes using the commands in the `Makefile`.
+
+For example to start the default agent in `docker`, run the following:
+
+```shell
+make agent/up
+```
+
+All artifacts created during the procedure are stored under the `.ipc` directory, which has the following structure:
+
+```console
+❯ tree -a .ipc
+.ipc
+├── agents
+│   └── agent-0
+│       ├── compose.yaml
+│       ├── config.toml
+│       ├── config.toml.orig
+│       ├── .env
+│       └── subnets
+│           └── node-0
+└── nodes
+    └── node-0
+        ├── compose.yaml
+        └── .env
+
+5 directories, 7 files
+```
+
+There can be multiple agents, and their corresponding `config.toml` files will be generated as we create more nodes and subnets. To create another agent, we would run `make agent IPC_AGENT_NR=1`.
+
+The main targets of the `Makefile` are:
+
+* `make agent`: create a configuration directory for `$IPC_AGENT_NR`; the container isn't started yet, so we could make some modifications if necessary
+* `make agent/up`: start the docker container for `$IPC_AGENT_NR`; if necessary build the `ipc-agent` docker image, the configuration directory, etc.
```

Comment:
Quick sanity-check: does the script check the commit it is on and re-generates the image automatically if the commit changes, or this should be manually by users? I am just wondering if we should add a note here around that.

Reply:
At he moment the dependency is here:
https://github.com/consensus-shipyard/ipc-agent/blob/1749ce37556631ee90498a353fcc1007797d71b9/testing/e2e/Makefile#L177

which depends on the agent image: 
https://github.com/consensus-shipyard/ipc-agent/blob/1749ce37556631ee90498a353fcc1007797d71b9/testing/e2e/Makefile#L270

Which depends on all the Rust source. 

Because the dependency is this form:
```
$(IPC_AGENT_DIR)/steps/up: .make/docker/network .make/docker/images/ipc-agent
```
and not this with the extra `|`:
```
$(IPC_AGENT_DIR)/steps/up: | .make/docker/network .make/docker/images/ipc-agent
```
I expect that it would indeed re-build the image and redeploy the agent.

Whether that's expected and valuable, I don't know. I think with the agent there's no harm in it, and if the same thing happened to the node, hopefully the mounted `repo_directory` volume would survive.